test/lib/graph_utils_test.rb

```diff
@@ -0,0 +1,30 @@
+require "test_helper"
+require 'graph_utils'
+
+class GraphUtilsTest < ActiveSupport::TestCase
+  def test_detect_cycles
+    assert GraphUtils.contains_cycle?([[1], [0]])
+    assert GraphUtils.contains_cycle?([[1], [2], [0]])
+    assert GraphUtils.contains_cycle?([[1], [2], [3], [1]])
```

Comment:
Can we have a `refute` or two in here too pls. Atm the test would pass if it `contains_cycle?` just returned `true`.

Reply:
done